lib/matplotlib/axes/_axes.py

```diff
@@ -2344,6 +2345,14 @@ def bar(self, x, height, width=0.8, bottom=None, *, align="center",
                 # Fallback if edgecolor == "none".
                 itertools.repeat('none'))
 
+        # Resolve the float - Decimal TypeError
```

Comment:
I think this is the wrong way to go about this.  You should convert x and y from Decimal at some (early) point.  I think at 
```python
x = np.asarray(self.convert_xunits(x), dtype=np.float)
```
will do the conversion.  Really convert_xunits should probably be guaranteed to return a float...

Reply:
Hi @jklymak ,

Thank you very much to point this out! Please forgive my lack of knowledge about matplotlib. This time I added a helper function that use np.asarray and/or ma.asarray for conversion, and added calls to it right after self.convert_xunits() and/or self.convert_yunits().

Reply:
Hi @jklymak,
I have followed all the following suggestions and updated my code.